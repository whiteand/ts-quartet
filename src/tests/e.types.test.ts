import { e as v } from "..";
import { testValidatorImpure } from "./testValidatorImpure";

describe("v.[type]", () => {
  test("v.boolean", () => {
    const checkBoolean = v(v.boolean);
    testValidatorImpure(checkBoolean, [true, false], ["true", "false", 1, 0]);
  });
  test("v.number", () => {
    const checkNumber = v(v.number);
    testValidatorImpure(
      checkNumber,
      [1, 0, 1.5, Infinity, -Infinity, -1, NaN],
      [
        "1",
        null,
        undefined,
        {},
        [],
        {
          valueOf() {
            return 1;
          }
        }
      ]
    );
  });
  test("v.string", () => {
    const checkString = v(v.string);
    testValidatorImpure(
      checkString,
      ["1", ""],
      [Symbol.for("quartet"), String, null, 0, undefined, true, false]
    );
  });
  test("v.finite", () => {
    const checkFinite = v(v.finite);
    testValidatorImpure(
      checkFinite,
      [1, 0, 1.5, -1],
      [
        "1",
        Infinity,
        -Infinity,
        NaN,
        null,
        undefined,
        {},
        [],
        {
          valueOf() {
            return 1;
          }
        }
      ]
    );
  });
  test("v.safeInteger", () => {
    const checkSafeInteger = v(v.safeInteger);
    testValidatorImpure(
      checkSafeInteger,
      [1, 0, -1],
      [
        "1",
        1.5,
        Infinity,
        -Infinity,
        NaN,
        null,
        undefined,
        {},
        [],
        {
          valueOf() {
            return 1;
          }
        }
      ]
    );
  });
  test("v.function", () => {
    const checkFunction = v(v.function);
    testValidatorImpure(
      checkFunction,
      [
        () => true,
        function() {
          return true;
        },
        new Function("return true")
      ],
      [1, null, undefined]
    );
  });
  test("v.symbol", () => {
    const checkSymbol = v(v.symbol);
    testValidatorImpure(
      checkSymbol,
      [Symbol.for("quartet"), Symbol.for("andrew"), Symbol("123")],
      ["symbol", null, undefined]
    );
  });
  test("v.positive", () => {
    const checkPositive = v(v.positive);
    testValidatorImpure(
      checkPositive,
      [
        1,
        1.5,
        Infinity,
        [1],
        "1",
        {
          valueOf() {
            return 1;
          }
        }
      ],
      [0, null, -Infinity, -1, NaN, undefined, {}, []]
    );
  });
  test("v.negative", () => {
    const checkNegative = v(v.negative);
    testValidatorImpure(
      checkNegative,
      [
        -1,
        -1.5,
        -Infinity,
        [-1],
        "-1",
        {
          valueOf() {
            return -1;
          }
        }
      ],
      [0, null, Infinity, 1, NaN, undefined, {}, []]
    );
  });
});
