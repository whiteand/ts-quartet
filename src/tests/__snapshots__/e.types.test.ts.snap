// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v.[type] v.boolean 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Boolean",
    },
    "value": "true",
  },
]
`;

exports[`v.[type] v.boolean 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Boolean",
    },
    "value": "false",
  },
]
`;

exports[`v.[type] v.boolean 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Boolean",
    },
    "value": 1,
  },
]
`;

exports[`v.[type] v.boolean 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Boolean",
    },
    "value": 0,
  },
]
`;

exports[`v.[type] v.finite 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": "1",
  },
]
`;

exports[`v.[type] v.finite 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": Infinity,
  },
]
`;

exports[`v.[type] v.finite 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": -Infinity,
  },
]
`;

exports[`v.[type] v.finite 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": NaN,
  },
]
`;

exports[`v.[type] v.finite 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.finite 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.finite 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": Object {},
  },
]
`;

exports[`v.[type] v.finite 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": Array [],
  },
]
`;

exports[`v.[type] v.finite 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Finite",
    },
    "value": Object {
      "valueOf": [Function],
    },
  },
]
`;

exports[`v.[type] v.function 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Function",
    },
    "value": 1,
  },
]
`;

exports[`v.[type] v.function 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Function",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.function 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Function",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.negative 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": 0,
  },
]
`;

exports[`v.[type] v.negative 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.negative 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": Infinity,
  },
]
`;

exports[`v.[type] v.negative 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": 1,
  },
]
`;

exports[`v.[type] v.negative 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": NaN,
  },
]
`;

exports[`v.[type] v.negative 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.negative 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": Object {},
  },
]
`;

exports[`v.[type] v.negative 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Negative",
    },
    "value": Array [],
  },
]
`;

exports[`v.[type] v.number 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": "1",
  },
]
`;

exports[`v.[type] v.number 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.number 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.number 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": Object {},
  },
]
`;

exports[`v.[type] v.number 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": Array [],
  },
]
`;

exports[`v.[type] v.number 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Number",
    },
    "value": Object {
      "valueOf": [Function],
    },
  },
]
`;

exports[`v.[type] v.positive 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": 0,
  },
]
`;

exports[`v.[type] v.positive 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.positive 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": -Infinity,
  },
]
`;

exports[`v.[type] v.positive 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": -1,
  },
]
`;

exports[`v.[type] v.positive 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": NaN,
  },
]
`;

exports[`v.[type] v.positive 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.positive 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": Object {},
  },
]
`;

exports[`v.[type] v.positive 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Positive",
    },
    "value": Array [],
  },
]
`;

exports[`v.[type] v.safeInteger 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": "1",
  },
]
`;

exports[`v.[type] v.safeInteger 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": 1.5,
  },
]
`;

exports[`v.[type] v.safeInteger 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": Infinity,
  },
]
`;

exports[`v.[type] v.safeInteger 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": -Infinity,
  },
]
`;

exports[`v.[type] v.safeInteger 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": NaN,
  },
]
`;

exports[`v.[type] v.safeInteger 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.safeInteger 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.safeInteger 8`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": Object {},
  },
]
`;

exports[`v.[type] v.safeInteger 9`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": Array [],
  },
]
`;

exports[`v.[type] v.safeInteger 10`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "SafeInteger",
    },
    "value": Object {
      "valueOf": [Function],
    },
  },
]
`;

exports[`v.[type] v.string 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": Symbol(quartet),
  },
]
`;

exports[`v.[type] v.string 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": [Function],
  },
]
`;

exports[`v.[type] v.string 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.string 4`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": 0,
  },
]
`;

exports[`v.[type] v.string 5`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": undefined,
  },
]
`;

exports[`v.[type] v.string 6`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": true,
  },
]
`;

exports[`v.[type] v.string 7`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "String",
    },
    "value": false,
  },
]
`;

exports[`v.[type] v.symbol 1`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Symbol",
    },
    "value": "symbol",
  },
]
`;

exports[`v.[type] v.symbol 2`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Symbol",
    },
    "value": null,
  },
]
`;

exports[`v.[type] v.symbol 3`] = `
Array [
  Object {
    "path": Array [],
    "schema": Object {
      "type": "Symbol",
    },
    "value": undefined,
  },
]
`;
